apps/explorer/src/components/ownedobjects/views/UnderlyingObjectCard.tsx

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useGetDynamicFieldObject } from '@mysten/core';
+import { getObjectDisplay, getObjectFields, getObjectId } from '@mysten/sui.js';
+
+import { SyntaxHighlighter } from '~/components/SyntaxHighlighter';
+import DisplayBox from '~/components/displaybox/DisplayBox';
+import { Banner } from '~/ui/Banner';
+import { ObjectLink } from '~/ui/InternalLink';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Text } from '~/ui/Text';
+import {
+    parseImageURL,
+    parseObjectType,
+    extractName,
+} from '~/utils/objectUtils';
+import { trimStdLibPrefix } from '~/utils/stringUtils';
+
+interface UnderlyingObjectCardProps {
+    parentId: string;
+    name: {
+        type: string;
+        value?: string;
+    };
+}
+
+export function UnderlyingObjectCard({
+    parentId,
+    name,
+}: UnderlyingObjectCardProps) {
+    const { data, isLoading, isError, isFetched } = useGetDynamicFieldObject(
+        parentId,
+        name
+    );
+
+    if (isLoading) {
+        return (
+            <div className="mt-3 flex w-full justify-center pt-3">
+                <LoadingSpinner text="Loading data" />
+            </div>
+        );
+    }
+
+    if (isError || data.error || (isFetched && !data)) {
+        return (
+            <Banner variant="error" spacing="lg" fullWidth>
+                Failed to get field data for :{parentId}
+            </Banner>
+        );
+    }
+    const display = getObjectDisplay(data);
+    const imgUrl = parseImageURL(display.data);
+    const objectType = parseObjectType(data);
+    const caption = extractName(display.data) || trimStdLibPrefix(objectType);
+    // For
+    const underlyingObjectTypes =
```

Comment:
I think we can always show the underlying object. Why to display it only when the name.value is object? All fields have a dynamic_field::Field that has a key with some type and a value with some type.